test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,177 @@
+	// Flush server and reset aggregators after the test is ran
+	if s.DevMode {
+		s.cleanUp()
+	}
+	s.Env().Fakeintake.FlushServerAndResetAggregators()
+}
+
+func (s *LinuxVMFakeintakeSuite) TestLinuxLogTailing() {
+	// Run test cases
+	s.Run("LogCollection", func() {
+		s.LogCollection()
+	})
+
+	s.Run("LogPermission", func() {
+		s.LogPermission()
+	})
+
+	s.Run("LogRotation", func() {
+		s.LogRotation()
+	})
+}
+
+func (s *LinuxVMFakeintakeSuite) LogCollection() {
+	t := s.T()
+	fakeintake := s.Env().Fakeintake
+
+	// Create a new log file
+	s.Env().VM.Execute("sudo touch /var/log/hello-world.log")
+
+	// Part 1: Ensure no logs are present in fakeintake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		logs, err := fakeintake.FilterLogs("hello")
+		if !assert.NoError(c, err, "Unable to filter logs by the service 'hello'.") {
+			return
+		}
+		assert.Empty(c, logs, "Logs were found when none were expected.")
+
+		// If logs are found, print their content for debugging
+		if len(logs) != 0 {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+			require.Empty(c, logs, "Logs were found when none were expected.")
+		}
+	}, 5*time.Minute, 10*time.Second)
+
+	// Part 2: Adjust permissions of new log file
+	_, err := s.Env().VM.ExecuteWithError("sudo chmod 777 /var/log/hello-world.log")
+	require.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Generate log
+	generateLog(s, "hello-world")
+
+	// Part 3: Assert that logs are found in intake after generation
+	checkLogs(s, "hello", "hello-world")
+}
+
+func (s *LinuxVMFakeintakeSuite) LogPermission() {
+
+	// Part 4: Block permission and check the Agent status
+	s.Env().VM.Execute("sudo chmod 000 /var/log/hello-world.log")
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		// Check the Agent status
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		if !assert.NoError(c, err, "Issue running agent status: %s", err) {
+			return
+		}
+
+		if strings.Contains(statusOutput, "Status: OK") {
+			assert.Fail(c, "log file is unexpectedly accessible")
+			return
+		}
+
+		require.Contains(c, statusOutput, "denied", "Log file is correctly inaccessible")
+	}, 3*time.Minute, 10*time.Second)
+
+	// Part 5: Restore permissions
+	s.Env().VM.Execute("sudo chmod 777 /var/log/hello-world.log")
+
+	// Part 6: Restart the agent, generate new logs
+	s.Env().VM.Execute("sudo service datadog-agent restart")
+
+	generateLog(s, "hello-world")
+
+	// Check the Agent status
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		require.NoError(c, err, "Issue running agent status: %s", err)
+		require.Contains(c, statusOutput, "Status: OK", "Expecting log file to be accessible but it is inaccessible instead")
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Until https://github.com/stretchr/testify/pull/1481 is merged and released, `require` panics on error, preventing the eventually to try again. You might want to change to `assert` or remove the eventually